src/Item/Item.service.ts

```diff
@@ -645,4 +626,25 @@ export class ItemService {
     }
     return Bridge.toFullItem(upsertedItem, dbCollection)
   }
+
+  private async checkIfThirdPartyItemURNExists(
+    id: string,
+    urn: string,
+    action = ItemAction.UPSERT
+  ): Promise<void> {
+    const decodedItemURN = decodeThirdPartyItemURN(urn)
+    if (
+      await Item.isURNRepeated(
+        id,
+        decodedItemURN.third_party_id,
+        decodedItemURN.item_urn_suffix
+      )
+    ) {
+      throw new URNAlreadyInUseError(id, urn, action)
+    }
+    const [wearable] = await peerAPI.fetchWearables<ThirdPartyWearable>([urn])
```

Comment:
It might be better to do this call alongside the `Item.isURNRepeated` one with a `Promise.all`.

Reply:
I think calling them in a sequence might be better in some scenarios, where we avoid calling the catalyst if we already know by consulting the database. What do you think?